aten/src/ATen/TensorIterator.cpp

```diff
@@ -475,17 +475,38 @@ void TensorIteratorBase::allocate_or_resize_outputs() {
         }
       }
       auto tensor_shape = invert_perm(shape_);
+
+      // For quantized output tensors, get the quantization params from the
+      // input.
+      c10::optional<TensorQuantizationOptions> quant_options = c10::nullopt;
+      const auto& options = op.options();
+      if (options.dtype() == at::kQInt8 || options.dtype() == at::kQUInt8 ||
+          options.dtype() == at::kQInt32) {
+        // get the first input and copy its quantization parameters
+        const auto& first_input = operands_[num_outputs_];
```

Comment:
So what if the quantization parameters on the two tensors are different? That seems like a totally reasonable situation, and then copying the quantization parameters from the first input seems like a totally arbitrary choice.

Reply:
If the correct quantization of the output tensor is something that varies from operator to operator, it may make sense to force kernels to do the computation of the quantization. There is some precedent for doing this in TensorIterator with the `static_shape` parameter, where kernels can say "Hey, I know you've got some built-in shape computation, but I actually happen to know what the output shape is, please use this." I could easily imagine something similar for quantization.

Reply:
yeah, I don't think the current solution is super clean, I'm also not sure if the alternative would be better.

The two high level ways I saw to do this were:
1. add a quantized path to the current fp32 indexing logic (current PR).  This reuses the size calculations and the kernels, but has to add some metadata for quantized tensors in places where it previously did not exist.
2. dispatch to something like `quantized_index`, which would have to reimplement the indexing logic (with some pieces potentially reusable from fp32, but likely not all)

Thoughts on any options I might be missing here?  Does 1 sound like the right call?

Reply:
for now, it's option 1, with the non-relevant parts asserted out, but I'm flexible

Reply:
@vkuzo The nuance here is that the code paths you are editing in TensorIterator aren't just for fp32 indexing: they are for *every unary, binary and reduction operation in PyTorch*. So if you want to add something here, there is an obligation to at least have some plausible argument why the logic here is appropriate for all other operations.

I don't mind option one, but this PR ain't it. An implementation that more closely follows the idea of (1) is to take the existing indexing kernel (not in TensorIterator) and program TensorIterator on what the intended quantization result is.